src/test/java/org/springframework/integration/dsl/test/file/FileTests.java

```diff
@@ -307,7 +309,11 @@ public IntegrationFlow tailFlow() {
 		@Bean
 		public IntegrationFlow fileReadingFlow() {
 			return IntegrationFlows
-					.from(s -> s.file(tmpDir.getRoot()).patternFilter("*.sitest"),
+					.from(s -> s.file(tmpDir.getRoot())
+									.patternFilter("*.sitest")
+									.useWatchService(true)
```

Comment:
Well, sounds like we have to expose an new `midifiers` option on the `FileReadingMessageSource`.
Although I don't like to rely on the `com.sun.*` package. But... that is for implementation.
Everything rest is up to end-user! :smile: